### ISSUE TITLE
[java] ImmutableField: False positive when assigning field value in branches of try-catch blocks

### ISSUE THREAD

**Issue by Nightfirecat** [pmd/pmd#6466]
**Affects PMD Version:** 7.2.0

**Rule:** [ImmutableField](https://docs.pmd-code.org/latest/pmd_rules_java_design.html#immutablefield)

**Description:**

When assigning a field value within the branches of a try-catch statement, PMD suggests making it final. However, doing so leads to a java compilation error, namely "Variable might already have been assigned to"

**Code Sample demonstrating the issue:**

```java
class TestClass {
    private String field; // PMD reports the false positive on this line

    TestClass() {
        try {
            field = "";
        } catch (Exception e) {
            field = "";
        }
    }
}
```

**Expected outcome:**

The false positive reported on line 2 above would not be reported

**Running PMD through:** *Gradle*

**Comment by UncleOwen** [comment]
Looks like this was fixed since then:
```
$ pmd-7.2.0 check --rulesets=immutablefield.xml --dir=. 
[WARN] Progressbar rendering conflicts with reporting to STDOUT. No progressbar will be shown. Try running with argument -r <file> to output the report to a file instead.
[WARN] This analysis could be faster, please consider using Incremental Analysis: https://docs.pmd-code.org/pmd-doc-7.2.0/pmd_userdocs_incremental_analysis.html
./src/TestClass.java:2: ImmutableField: Field 'field' may be declared final
$ pmd-7.23.0 check --rulesets=immutablefield.xml --dir=. 
[WARN] Progressbar rendering conflicts with reporting to STDOUT. No progressbar will be shown. Try running with argument -r <file> to output the report to a file instead.
[WARN] This analysis could be faster, please consider using Incremental Analysis: https://docs.pmd-code.org/pmd-doc-7.23.0/pmd_userdocs_incremental_analysis.html
```
